mmengine/_strategy/base.py

```diff
@@ -0,0 +1,870 @@
+        """Load scheduler state from dict."""
+        if isinstance(self.param_schedulers, dict):
+            assert isinstance(state_dict, dict)
+            for name, schedulers in self.param_schedulers.items():
+                for scheduler, ckpt_scheduler in zip(schedulers,
+                                                     state_dict[name]):
+                    scheduler.load_state_dict(ckpt_scheduler)
+        else:
+            for scheduler, ckpt_scheduler in zip(
+                    self.param_schedulers,  # type: ignore
+                    state_dict):
+                scheduler.load_state_dict(ckpt_scheduler)
+
+    def load_or_resume(self) -> Optional[dict]:
+        """Load checkpoint or resume from checkpoint."""
+        from mmengine.runner import find_latest_checkpoint
+
+        if not self._resume and self._load_from is None:
+            return None
+
+        # decide to load from checkpoint or resume from checkpoint
+        resume_from = None
+        if self._resume and self._load_from is None:
+            # auto resume from the latest checkpoint
+            resume_from = find_latest_checkpoint(self._work_dir)
+            self.logger.info(
+                f'Auto resumed from the latest checkpoint {resume_from}.')
+        elif self._resume and self._load_from is not None:
+            # resume from the specified checkpoint
+            resume_from = self._load_from
+
+        if resume_from is not None:
+            return self.resume(resume_from)
+        elif self._load_from is not None:
+            return self.load_checkpoint(self._load_from)
+
+        return None
+
+    @abstractmethod
+    def load_checkpoint(
+        self,
+        filename: str,
+        *,
+        map_location: Union[str, Callable] = 'cpu',
+        callback: Optional[Callable] = None,
+    ) -> dict:
+        """Load checkpoint from given ``filename``.
+
+        Args:
+            filename (str): Accept local filepath, URL, ``torchvision://xxx``,
+                ``open-mmlab://xxx``.
+            map_location (str or callable): A string or a callable function to
+                specifying how to remap storage locations.
+                Defaults to 'cpu'.
+        """
+
+    @abstractmethod
+    def resume(
+        self,
+        filename: str,
+        *,
+        resume_optimizer: bool = True,
+        resume_param_scheduler: bool = True,
+        map_location: Union[str, Callable] = 'default',
+        callback: Optional[Callable] = None,
+    ) -> dict:
+        """Resume training from given ``filename``.
+
+        Four types of states will be resumed.
+
+        - model state
+        - optimizer state
+        - scheduler state
+        - randomness state
+
+        Args:
+            filename (str): Accept local filepath, URL, ``torchvision://xxx``,
+                ``open-mmlab://xxx``.
+        """
+
+    @abstractmethod
+    def save_checkpoint(
```

Comment:
How about renaming `save_checkpoint` to `save`? Otherwise it will be strange that calling `save_checkpoint` without calling `load_checkpoint`.